lib/commands/init.js

```diff
@@ -78,6 +86,14 @@ module.exports = Command.extend({
       return Promise.reject(new SilentError(`We currently do not support a name of \`${packageName}\`.`));
     }
 
+    // [WIP] ember new foo --lang
+    let langDef = this.availableOptions.find((opt) => opt.name === 'lang').default;
+    let langInfo = getLangArgInfo(commandOptions.lang);
+    if (!langInfo.result && langInfo.langArgValue !== langDef) {
+      return Promise.reject(new SilentError(langInfo.message));
+    }
+    commandOptions.lang = langInfo.langArgValue;
```

Comment:
I would think that this would be sufficient:

```suggestion
    if (commandOptions.lang) {
      let langInfo = getLangArgInfo(commandOptions.lang);
      commandOptions.lang = langInfo.langArgValue;
    }
```

RE: using the custom `langInfo.message`, I would expect `getLangArgInfo` to directly `throw new SilentError(langInfo.message)` internally.

Reply:
This has been partially resolved in f95d3a9 in that the logic here has been updated. The error handling hasn't been moved to `getLangArgInfo` yet, but in the meantime, it has been updated per the comment about using `throw`.

Reply:
Update:  the error handling has been moved to the utility (and removed from `init`) as part of b3f03f4 👍